pyqtgraph/colormap.py

```diff
@@ -606,9 +604,9 @@ def map(self, data, mode=BYTE):
         # Convert to QColor if requested
         if mode == self.QCOLOR:
             if np.isscalar(data):
-                return QtGui.QColor(*interp)
+                return QtGui.QColor.fromRgbF(*interp)
             else:
-                return [QtGui.QColor(*x.tolist()) for x in interp]
+                return [QtGui.QColor.fromRgbF(*x.tolist()) for x in interp]
         else:
```

Comment:
Thank you, that is very much what I had in mind!

I am glad that you implemented this; I am sorry that I didn't find any time to add this after you pointed out the factory functions.